### DISCUSSION TITLE
HttpPost not working in .Net, Orchard CMS project

### DISCUSSION THREAD

**Discussion by sushmita-inamdar** [OrchardCMS/OrchardCore#16370] [Q&A] [answered]
Hi,

I have several GET requests in my .Net, Orchard project and I wanted to add a **POST** request as below - 
```

[HttpPost("parking/occupancy")]
public async Task<IActionResult> Occupancies([FromBody] List<string> ids)
{
    var results = new List<ParkingOccupancy>();

    foreach (var id in ids)
    {
        var parkingLot = await _contentManager.GetAsync(id);
        if (parkingLot == null || parkingLot.ContentType != "ParkingLot" || !await AuthorizeAsync(parkingLot))
        {
            return NotFound();
        }

        var idToGarage = await GetGarageIdsToGaragesAsync();
        var percentFull = GetPercentFullAsync(parkingLot, idToGarage);

        results.Add(new ParkingOccupancy
        {
            SysId = id,
        });
    }

    return Ok(results);
}

```
The above method should take a list of ids as input and fetches the parking occupancies for each of those ids.
But for some reason, the POSTMAN input is not even hitting the above method on debug and gives out the below error - 

![image](https://github.com/OrchardCMS/OrchardCore/assets/159943385/f3b0770f-db01-49a8-ac6a-5895d00717b6)

**Is there any modifications or any code setting changes required to handle POST requests in my project?**

**Comment by weirdyang** [comment]
Are these controllers in a module or in the web application itself?

**Reply by sushmita-inamdar** [reply]
Hi @weirdyang,
This controller is within a module.

**Reply by weirdyang** [reply, marked as answer]
probably an authentication issue, try adding these attributes to the controller to confirm

`[IgnoreAntiforgeryToken, AllowAnonymous]`

**Reply by sushmita-inamdar** [reply]
@weirdyang Hey, it worked! Thank you so much.
